### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,64 @@
...
+    def __le__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, int | float):
+            return self.km <= other
+        else:
+            return self.km <= other.km
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, int | float):
+            return self.km >= other
+        else:
```
Comment (Vasyl-Smutok) [inline app/main.py:63]: `else` is redundant here in other places

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
```
Comment (olena-hul) [inline app/main.py:5]: int is also possible here

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+        return f"Distance(km={self.km})"
+
+    def __add__(
+            self, other: Distance | int | float
+    ) -> Distance:
+        if isinstance(other, int | float):
+            self.km = self.km + other
+        else:
+            self.km = self.km + other.km
+        return self
```
Comment (olena-hul) [inline app/main.py:21]: `__add__` should return a new Distance instance, it should not override km attribute

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+
+    def __iadd__(self, other: Distance) -> Distance:
+        result = self + other
+        return result
```
Comment (olena-hul) [inline app/main.py:25]: Here should be the code of `__add__` method)

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by Vasyl-Smutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**